src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -514,20 +512,82 @@ else if (fileProcessorName.get() instanceof List<?>)
 						}
 					}
 				}
-
+				if (refCreator instanceof NCBIGeneBasedReferenceCreator || refCreator instanceof ENSMappedIdentifiersReferenceCreator)
+				{
+					for (EntrezGeneBasedReferenceCreator subCreator : ((NCBIGeneBasedReferenceCreator)refCreator).getSubCreators())
```

Comment:
I'm a bit confused about why `ENSMappedIdentifiersReferenceCreator` is checked as a possible instance type if the refCreator is cast to `NCBIGeneBasedReferenceCreator` in the for loop?

Reply:
I could have cast to either `ENSMappedIdentifiersReferenceCreator` or `ENSMappedIdentifiersReferenceCreator`. Either one should work, but I have to choose one.

Reply:
...but if I _don't_ cast, then I don't get access to `getSubCreators`.

Reply:
My question here is more why is `ENSMappedIdentifiersReferenceCreator` checked as a possibility if the cast on line 122 is going to be `NCBIGeneBasedReferenceCreator`?  Would the first condition on line 520 `refCreator instanceof NCBIGeneBasedReferenceCreator` be enough and the `||` not needed?

Reply:
There is no cast on line 122... :/

Reply:
Because `refCreator` could be either an `NCBIGeneBasedReferenceCreator`, _or_ it could be an `ENSMappedIdentifiersReferenceCreator`. When it comes time to use it, it doesn't matter _which_ we cast to - as long as we cast to _something_ that will give access to the `getSubCreators` method.

Reply:
Yeah, line 522 - sorry.  
That's true, but isn't `ENSMappedIdentifiersReferenceCreator` a sub-class of `NCBIGeneBasedReferenceCreator`?  So you only need to have the more general check of `refCreator instanceof NCBIGeneBasedReferenceCreator` on line 520.  If that's false, then the second check will never be true.

Reply:
That is my issue isn't with the cast - it's with the second `instanceof` check.

Reply:
Hmm.... I believe this was was necessary in the past but I think the type hierarchy has changed so it may no longer be necessary. I could remove it, but that would require testing. Might not finish by the end of the day...

Reply:
Okay, it's not a blocker but just something I noticed that could be simplified.  I don't want to hold up approving and merging the PR for it, so maybe just a TODO comment to test it?

Reply:
Testing is already underway.